GitUI/CommandsDialogs/FormResolveConflicts.cs

```diff
@@ -1007,41 +1148,71 @@ private bool CheckForRemoteRevision(ConflictData item)
                 return true;
             }
 
-            string caption = string.Format(_fileModifiedLocallyAndDeletedRemotelyLong.Text,
-                item.Filename,
-                GetLocalSideString(),
-                GetRemoteSideString());
-
-            using (var frm = new FormModifiedDeletedCreated(string.Format(_keepModifiedButtonText.Text + " ({0})", GetLocalSideString()),
-                string.Format(_deleteFileButtonText.Text + " ({0})", GetRemoteSideString()),
-                _keepBaseButtonText.Text,
-                caption))
+            string dialogText = "";
+            if (_solveMergeConflictApplyToAll == false && _filesModifiedLocallyAndDeletedRemotelySolved == 1)
             {
-                frm.ShowDialog(this);
+                dialogText = string.Format(_fileModifiedLocallyAndDeletedRemotelyLong.Text,
+                    item.Filename,
+                    GetLocalSideString(),
+                    GetRemoteSideString());
+            }
 
-                if (frm.KeepBase)
+            if (_solveMergeConflictApplyToAll == false && _filesModifiedLocallyAndDeletedRemotelySolved > 1)
+            {
+                if (_filesDeletedLocallyAndModifiedRemotelyCount == 0 && _filesRemainedCount == 0)
                 {
-                    // base
-                    ChooseBaseOnConflict(item.Filename);
+                    // Task Dialog shows the name of a current file, so the amount of the left files should be minus one. That's why _filesModifiedLocallyAndDeletedRemotelySolved - 1
+                    dialogText = string.Format(_filesModifiedLocallyAndDeletedRemotelyLongNoOtherFilesSelected.Text, item.Filename, _filesModifiedLocallyAndDeletedRemotelySolved - 1);
                 }
-
-                if (frm.KeepLocal)
+                else
                 {
-                    // delete
-                    ChooseLocalOnConflict(item.Filename);
+                    dialogText = string.Format(_filesModifiedLocallyAndDeletedRemotelyLong.Text, item.Filename, _filesModifiedLocallyAndDeletedRemotelySolved - 1, _conflictItemsCount);
                 }
+            }
+
+            string dialogInstruction = string.Empty;
+            string dialogCaption = _solveMergeConflictDialogCaption.Text;
+            string dialogFooterCheckboxText = (_filesModifiedLocallyAndDeletedRemotelySolved > 1) ? string.Format(_solveMergeConflictApplyToAllCheckBoxText.Text, item.Filename, _filesModifiedLocallyAndDeletedRemotelySolved - 1) : string.Empty;
+
+            string keepLocalButtonText = string.Format(_keepModifiedButtonText.Text + " ({0})", GetLocalSideString());
```

Comment:
This (and similar) is done in a number of places for seemingly no good reason - we're creating many unnecessary string objects.
```suggestion
            string keepLocalButtonText = string.Concat(_keepModifiedButtonText.Text, $" (GetLocalSideString())");
```

Reply:
Okay, I will remove local string objects and pass the values directly as arguments